pycozmo/emotions.py

```diff
@@ -18,17 +20,48 @@
 ]
 
 
+class Node:
+    def __init__(self, x: float, y: float):
+        self.x = x
+        self.y = y
+
+
+class DecayGraph:
+    def __init__(self, nodes: List[Node]):
+        self.nodes_x = [node.x for node in nodes]
+        self.nodes_y = [node.y for node in nodes]
+        if len(nodes) > 1:
+            self.ext_line_params = self.get_line_parameters(nodes[-2], nodes[-1])
+
+    def get_increment(self, val) -> float:
+        if len(self.nodes_x) == 1:
+            f_out = self.nodes_y
+        elif val <= self.nodes_x[-1]:
+            f_out = np.interp(val, self.nodes_x, self.nodes_y)
+        else:
+            f_out = self.ext_line_params[0] * val + self.ext_line_params[1]
+        return round(1 - f_out, 2)
```

Comment:
Fancy

Reply:
This rounding is mostly for the tests haha, should I move it away?

Reply:
I think I'll move it to the tests, doesn't make much sense here :sweat_smile: